gustaf/spline/base.py

```diff
@@ -547,6 +527,121 @@ def nurbs(self):
+        as_boundary : bool
+          Multipatch is a boundary object of a higher dimensional geometry. If
+          set to true, additional checks are performed on the interfaces,
+          requiring strict inter connectivity between all patches
+
+        Returns
+        -------
+        None
+        """
+        splinepy.Multipatch.__init__(
+            self,
+            splines=splines,
+            interfaces=interfaces,
+        )
+        GustafBase.__init__(self)
+        self._show_options = self.__show_option__(self)
+
+    @property
+    def show_options(self):
+        """
+        Show option manager for multipatch systems.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        show_options: MultipatchShowOption
+        """
+        return self._show_options
+
+    def show(self):
+        """Show the multipatch system."""
+        # Retrieve show options out of valid options
+        spline_list: List[GustafSpline] = self.splines.copy()
+        if self.show_options.get("boundary_ids", False):
+            bsp = self.boundary_patches().splines
+            bsp_id = np.abs(self.interfaces[self.interfaces < 0])
+            n_unique = len(np.unique(bsp_id))
+            colors = [
+                tuple(np.array(colorsys.hsv_to_rgb(i / n_unique, 1, 1)) * 255)
+                for i in range(n_unique)
+            ]
+            np.random.shuffle(colors)
+            for id, patch in zip(bsp_id, bsp):
+                patch.show_options["c"] = colors[bsp_id[id] - 1]
+            spline_list.extend(bsp)
+        # field data adaptation for multipatch
+        field_data = self.show_options.get("field_function", False)
+        v_min, v_max = np.inf, -np.inf
+        if field_data:
+            if type(field_data) is callable:
+                for spline in self.splines:
+                    res = enforce_len(
+                        spline.show_options.get("resolutions", 100),
+                        spline.para_dim,
+                    )
+                    spline.spline_data["field_data"] = SplineDataAdaptor(
+                        spline, function=field_data
+                    )
+                    a = spline.spline_data.as_scalar("field_data", res)
+                    v_min = min(v_min, np.min(a))
+                    v_max = max(v_max, np.max(a))
+            elif type(field_data) is str and field_data == "me":
+                for spline in self.splines:
+                    res = enforce_len(
+                        spline.show_options.get("resolutions", 100),
+                        spline.para_dim,
+                    )
+                    spline.spline_data["field_data"] = spline
+                    spline.show_options["field_name"] = "field_data"
+                    a = spline.spline_data.as_scalar("field_data", res)
+                    v_min = min(v_min, np.min(a))
+                    v_max = max(v_max, np.max(a))
+            else:
```

Comment:
The field option here to evaluate the coordinates of the multipatch spline on the multipatch.fields splines. 

ignore None splines?

Reply:
I still don't understand completely what you mean here. Is this just for the case that the multipatch?fields option is supposed to be plotted or for any case?

Reply:
E.g. in the part where the microstructures are computed, the fields that represent the derivatives of the microstructures are formed through functional composition, which is a very costly operation. In order to save memory and also to reduce the number of required computations, all splines that would evaluate to 0 are skipped and stored as None. So whenever a spline has a field which is just a NoneType-Object, the field that it is supposed to represent is a spline of values 0. That is also handy if a multipatch field only has local support